uint32 num_requested
---
uint8 depth_per
float32[] data
uint32 arr_dim1
uint32 arr_dim2
uint32 arr_dim3